pkg/solana/logpoller/worker.go

```diff
@@ -0,0 +1,372 @@
+package logpoller
+
+import (
+	"context"
+	"crypto/rand"
+	"fmt"
+	"math/big"
+	"sync"
+	"sync/atomic"
+	"time"
+
+	"github.com/smartcontractkit/chainlink-common/pkg/logger"
+	"github.com/smartcontractkit/chainlink-common/pkg/services"
+)
+
+var (
+	ErrProcessStopped   = fmt.Errorf("worker process has stopped")
+	ErrContextCancelled = fmt.Errorf("worker context cancelled")
+)
+
+const (
+	// DefaultMaxRetryCount is the number of times a job will be retried before being dropped.
+	DefaultMaxRetryCount = 6
+	// DefaultNotifyRetryDepth is the retry queue depth at which the worker group will log a warning.
+	DefaultNotifyRetryDepth = 200
+	// DefaultNotifyQueueDepth is the queue depth at which the worker group will log a warning.
+	DefaultNotifyQueueDepth = 100
+	// DefaultWorkerCount is the default number of workers in a WorkerGroup.
+	DefaultWorkerCount = 10
+)
+
+type worker struct {
+	Name  string
+	Queue chan *worker
+	Retry chan Job
+	Lggr  logger.Logger
+}
+
+func (w *worker) Do(ctx context.Context, job Job) {
+	if ctx.Err() == nil {
+		if err := job.Run(ctx); err != nil {
+			w.Lggr.Infof("job %s failed with error; retrying: %s", job, err)
+			w.Retry <- job
+		}
+	}
+
+	// put itself back on the queue when done
+	select {
+	case w.Queue <- w:
+	default:
```

Comment:
Not sure this is necessary. The worker queue is always the size of the number of workers. The default in this case would only run if the queue of workers was full, which it can't be unless we have accidentally created too many workers.

Reply:
> unless we have accidentally created too many workers.

Seems better to log an error just in case, so we know what's happening